lib/helpers/utils.js

```diff
@@ -3413,6 +3414,115 @@ export function parseMavenTree(rawOutput, pomFile) {
+  const versionRegex =
+    /^(?:\u001b\[\d+m.+ -> )?(?<version>[^\ \u001b]+)(?:.*\u001b\[0m)?/m;
+  const levelCache = new Map();
+  const moduleComponent = completeComponent({
+    name: module,
+    version: "latest",
+  });
+  dependencies.set(moduleComponent["bom-ref"], moduleComponent);
+  relations.set(moduleComponent["bom-ref"], []);
+  levelCache.set(0, moduleComponent["bom-ref"]);
+  let moduleFilePath = module;
+  const versionNumbers = [];
+  let indexOfBracket = moduleFilePath.indexOf("[");
+  for (let versionIndex = 0; indexOfBracket !== -1; versionIndex++) {
+    // Special handling for modules called something like 'main.init.sbt.models[2.12.20]'
+    // This needs to be turned into a path like 'main/init/sbt/models/2.12.20'
+    // However, since all other periods need to be changed to slashes, this needs something more...
+    versionNumbers.push(
+      moduleFilePath.substring(
+        indexOfBracket + 1,
+        moduleFilePath.indexOf("]", indexOfBracket + 1),
+      ),
+    );
+    moduleFilePath = moduleFilePath.replace(
+      `[${versionNumbers[versionIndex]}]`,
+      "[]",
+    );
+    indexOfBracket = moduleFilePath.indexOf("[", indexOfBracket + 1);
+  }
+  moduleFilePath = moduleFilePath.replaceAll(".", "/");
+  for (const versionNumber of versionNumbers) {
+    // Now put the versions we removed above back into the path and replace the brackets at the same time
+    moduleFilePath = moduleFilePath.replace("[]", `/${versionNumber}/`);
+  }
+  const dependencyTreeLog = readFileSync(
+    resolve(millRootPath, "out", moduleFilePath, "ivyDepsTree.log"),
+    "utf-8",
+  );
+  const dependencyTreeLines = dependencyTreeLog
+    .trim()
+    .split("\n")
+    .map((dependency) => dependency.replaceAll("\r", ""));
+  for (const line of dependencyTreeLines) {
+    const match = treeRegex.exec(line);
+    if (match === null) {
+      continue;
+    }
+    const level = match.groups.treeIndentation.length / 3;
+    let group;
+    let name;
+    let version;
+    if (match.groups.dependency.indexOf(":") === -1) {
+      name = match.groups.dependency;
+      version = "latest";
+    } else {
+      [group, name, version] = match.groups.dependency.split(":");
+      version = versionRegex.exec(version).groups.version;
+    }
+    const component = completeComponent({
+      group,
+      name,
+      version,
+    });
+    if (!dependencies.has(component["bom-ref"])) {
+      dependencies.set(component["bom-ref"], component);
+      relations.set(component["bom-ref"], []);
+    }
+    if (
+      !relations.get(levelCache.get(level - 1)).includes(component["bom-ref"])
+    ) {
+      relations.get(levelCache.get(level - 1)).push(component["bom-ref"]);
+    }
+    levelCache.set(level, component["bom-ref"]);
+  }
+  return moduleComponent["bom-ref"];
+}
+
+function completeComponent(component) {
+  component["type"] = component.group ? "library" : "application";
```

Comment:
Can we set evidence and properties such as `SrcFile`?

Reply:
We can also add support for license fetch by invoking `getMvnMetadata` before the return and making the functions `async`.

Reply:
> Can we set evidence and properties such as `SrcFile`?

Sure, I'll add that in -- I had planned to, but I guess I forgot while working on some other part of the implementation. 🤔

Reply:
> We can also add support for license fetch by invoking `getMvnMetadata` before the return and making the functions `async`.

Licenses are already fetched if you set the EnvVar -- all Java-based scanners run through the code when the EnvVar ist set!

Reply:
`SrcFile`-property added, evidence for the `purl` added.

Running `--evidence` gives me an error though, I guess this needs to be implemented in `atom`?
```
Creating reachables slice for /Users/malice/eclipse/workspace/mill. Please wait ...
Executing atom reachables -l mill -o /Users/malice/eclipse/workspace/mill/app.atom --slice-outfile /Users/malice/eclipse/workspace/mill/reachables.slices.json /Users/malice/eclipse/workspace/mill
This language is not yet supported by atom.
Failure: No language frontend supported for language 'MILL'
```